lib/java_buildpack/framework/jrebel_agent.rb

```diff
@@ -24,36 +24,51 @@ module Framework
     # Encapsulates the functionality for enabling zero-touch JRebel support.
     class JrebelAgent < JavaBuildpack::Component::VersionedDependencyComponent
 
+      def initialize(context, &version_validator)
+        super(context, &version_validator)
+        @component_name = 'JRebel Agent'
+      end
+
       # (see JavaBuildpack::Component::BaseComponent#compile)
       def compile
-        download_zip false
-        FileUtils.mv(download_location + 'jrebel.jar', @droplet.sandbox + jar_name)
-        FileUtils.remove_dir(download_location, true)
+        download_zip(true, @droplet.sandbox + 'install')
       end
 
       # (see JavaBuildpack::Component::BaseComponent#release)
       def release
         @droplet.java_opts
-          .add_javaagent(@droplet.sandbox + jar_name)
-          .add_bootclasspath_p(@droplet.sandbox + jar_name)
+          .add_agentpath(@droplet.sandbox + ('install/lib/' + lib_name))
           .add_system_property('rebel.remoting_plugin', true)
+          .add_system_property('rebel.log', true)
+          .add_system_property('rebel.base', @droplet.sandbox + '.jrebel')
+          .add_system_property('rebel.cloud.platform', 'cloudfoundry/java-buildpack')
       end
 
       protected
 
       # (see JavaBuildpack::Component::VersionedDependencyComponent#supports?)
       def supports?
-        jrebel_configured?(@application.root) || jrebel_configured?(@application.root + 'WEB-INF/classes')
+        jrebel_configured?(@application.root) || jrebel_configured?(@application.root + 'WEB-INF/classes') ||
+          jars_with_jrebel_configured?(@application.root)
       end
 
       private
 
       def jrebel_configured?(root_path)
-        (root_path + 'rebel.xml').exist? && (root_path + 'rebel-remote.xml').exist?
+        (root_path + 'rebel-remote.xml').exist?
+      end
+
+      def jars_with_jrebel_configured?(root_path)
+        jars = (root_path + '**/*.jar').glob
```

Comment:
This whole section should be collapsable with the following Ruby-ism:

```ruby
(root_path + '**/*.jar').glob.any? { |jar| ! `unzip -l "#{jar}" | grep "rebel-remote\\.xml$"`.strip.empty? }
```

Or something like that.  I've done that freehand, so you'll definitely want to test it. ;)